### ISSUE TITLE
Intervals BED file processing: remove headers and validate any timing information

### ISSUE THREAD

**Issue by tdanhorn** [nf-core/sarek#1148]
### Description of feature

The BED file (or, alternatively,  `*.intervals_list` file  as used by Picard Tools) file that is being passed to sarek (since version 3.0) with the `--intervals` parameter serves two functions:
* Define the intervals of interest, e.g. for performing BQSR and targeted variant calling
* Give hints to the scatter & gather algorithm on who to set the scatter intervals for optimal processing

Because the latter function allows the user to provide timing information in the 5th column (originally designated for a "score"), this can lead to spectacular derailments of the scattering/gathering if the BED/intervals file contains additional columns with name/score/strand or possibly annotation information, which is being interpreted as timing hints.  (I experienced a case where that lead to a splitting into several hundreds of thousands scatter intervals, instead of dozens at most, completely paralyzing the pipeline.  In that sense this might be considered a bug report.)

An unrelated, but equally annoying issue is that header lines in a BED added for display in the UCSC genome browser (see https://www.genome.ucsc.edu/FAQ/FAQformat.html#format1) lead to a crash, since they are "invalid" when being interpreted as BED lines denoting intervals (i.e. as `chr\tstart\tend`).  So if I have a BED file (e.g. provided by a WES kit manufacturer for identifying targets) that has more than 3 columns and/or one or more header lines, I would have to preprocess this file before using it, which is an extra step that may not be intuitive for new users and one more avoidable thing to do for anyone, never mind that another opportunity for making mistakes.

The provided BED/intervals file is being processed by the pipeline anyway (e.g. the staged copy is gzipped, and an `*.intervals_list` is converted to BED), so this problem can potentially be solved by making this preprocessing step "smarter".  This would involve the following steps:
* Scan the start of the provided BED for lines that match the regex `^(browser|track) ` (note the space at the end) and remove matching lines (this doesn't need to be done with a regex, since it is pretty simple).  This solves the crash problem when headers are present (`*.intervals_list` files don't have this problem and seems to be processed appropriately).
* Check if a 5th column is present, and if so, if it looks like timing information.  (Sorry, I've never used that feature, so I can't say exactly what would be valid or invalid for that, but it presumably would be strictly numeric.  Possibly a sanity check could be run to see if the numbers are plausible and lead to a manageable set of scatter intervals.)  If that is the case, use it as timing information, otherwise discard any columns after the 3rd (any names and scores don't matter, and strand information does not make sense in this context for DNA).

Thank you!

**Comment by heuermh** [comment]
> Because the latter function allows the user to provide timing information in the 5th column (originally designated for a "score"), this can lead to spectacular derailments of the scattering/gathering if the BED/intervals file contains additional columns with name/score/strand or possibly annotation information, which is being interpreted as timing hints.

Yeah, this is an abuse of the BED format. Please, let's not do this.
https://samtools.github.io/hts-specs/BEDv1.pdf

Timing information should be provided in a separate file.
